Source/GSNativeProtocol.m

```diff
@@ -493,24 +502,24 @@ - (NSURLResponse*) validateHeaderCompleteTransferState: (GSTransferState*)ts
 
 - (void) notifyDelegateAboutReceivedData: (NSData*)data
 {
-  NSURLSessionTask              *task;
-  id<NSURLSessionDelegate>      delegate;
+  NSURLSession                 *session;
+  NSURLSessionTask             *task;
+  id<NSURLSessionDelegate>     delegate;
 
   task = [self task];
-
   NSAssert(nil != task, @"Cannot notify");
 
+  session = [task session];
+  NSAssert(nil != session, @"Missing session");
```

Comment:
This is a change of semantic. Up to now the session was not required (although the code looks like it was). Now it is. Is this really intended?

Reply:
The original implementation declared the session variable inside the if block. I moved it outside to access it in the second if block.

Reply:
Yes, I know.  The difference is when there is no session in the old code there would be no delegate and for that reason the assert would not be triggered. Now when there is no session the assert gets triggered. The check in the old code never could be reached.